test/engine_test.go

```diff
@@ -101,3 +103,37 @@ func TestPreCompileHook(t *testing.T) {
 		t.Error("callback not called")
 	}
 }
+
+// TODO @ivokob is this test necessary, considering that the rangechecker is already tested against the test engine?
```

Comment:
Indeed I think this test is not particularly necessary. But I would add a test which would assert that the test engine implements `frontend.RangeChecker` to ensure that we won't have regression in the future.